x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -243,6 +244,22 @@ private static String[] enterpriseSearchAcknowledgementMessages(OperationMode cu
         return Strings.EMPTY_ARRAY;
     }
 
+    private static String[] esqlAcknowledgementMessages(OperationMode currentMode, OperationMode newMode) {
+        switch (newMode) {
```

Comment:
This looks clearer now but I still would feel better if it had a comment describing what the intent of it is.